### ISSUE TITLE
我有个问题想请教一下

### ISSUE THREAD

**Issue by markofan** [skyduy/CNN_keras#8]
我用keras搭建了一个CNN网络，我在model.fit()中设置了我的validation_data, 用的是vadidation_spilt=0.25
但是我运行的时候发现，training data的accuracy是80%左右，但是一开始我的val_acc就有100%了，而且一直保持不变，我想问一下，可能哪些地方出了问题？

**Comment by skyduy** [comment]
代码贴一下吧

**Comment by xdkaka** [comment]
这个问题是不是训练数据的问题，比如训练数据和预测数据有很大的重合度

**Comment by skyduy** [comment]
你数据从哪来的？一般情况下，既然分了train data和validation data，不会出现训练数据和验证数据有很大的重合度，还有val_acc竟然比train_acc高。。而且是达到100... 感觉是你validation acc计算代码出bug了。。。

**Comment by markofan** [comment]
对啊，我也不知道是哪里出现了问题。
我的model应该是没有问题的，因为我就用的师姐的model，唯一不同的是我的数据和我读数据的方法和师姐的不一样。
我的数据是MRI图像，它是3D的，我将很多张MRI图像数据读入一个mat文件，所以这个mat文件是4维的[428,69,59,57],其中428是图片的个数，后面3个是图片的长，宽，高。
我采用的是5折交叉验证的方法，首先采用的是sklearn的StratifiedKFold函数，将总的数据分成5份，其中一份为测试集，其他的4份为训练集。代码如下;
skf = StratifiedKFold(n_splits=5)
for train_index, test_index in skf.split(X_train, X_train_n):
    print('========== Now running on fold '+ str(i) + ' ==========')

    Y_train, Y_test=X_train[train_index],X_train[test_index]
    Y_train_n, Y_test_n=X_train_n[train_index],X_train_n[test_index]
验证集我是在fit函数中设置的，设置它为训练集的25%，代码如下：
logif = model.fit(Y_train, Y_train_n, batch_size=bsize, nb_epoch=nb_epoch, shuffle = False,
	   verbose=1,validation_split =0.25,callbacks =[csv_logger,reduce_lr,checkpointer,early_stopping])
其中的validation_split =0.25就是我的验证集。

**Comment by markofan** [comment]
验证集的accuracy计算应该也是在fit()函数中直接计算的吧？

**Comment by markofan** [comment]
而且我前3个folder的val_acc一直保持1,后面两个folder的val_acc却一直保持0.0581

**Comment by markofan** [comment]
我好像发现了一个问题：我的数据共有428个，其中有199个是不正常的，228个是正常的。所以其中两类分别占0.4649和0.535左右，我所选择的数据交叉验证，将数据分为训练集和测试集，训练集和测试集中两类的比例和总的数据是一样的（即在两个集中不正常占0.4649左右，正常的是0.535）,我现在将测试集的数据直接赋给验证集，val_data=test_data,此时得到的val_acc=0.5349,我认为就是原来数据的比重，感觉这个网络都没有训练，而我的traing_acc又有80%多。
我感觉是我的网络没有训练，这是什么问题？

**Comment by skyduy** [comment]
估计你validate的时候用的不是正在training的model。。。

**Comment by markofan** [comment]
那怎么才是用的正在训练的model。。。。

**Comment by markofan** [comment]
兄弟，能帮我看看代码不？
